src/main/java/net/smartcosmos/cluster/auth/SmartCosmosAuthenticationProvider.java

```diff
@@ -267,37 +232,13 @@ public UserDetails loadUserByUsername(String username) throws UsernameNotFoundEx
             userResponse.getTenantUrn(),
             userResponse.getUserUrn(),
             userResponse.getUsername(),
-            getPasswordHash(userResponse),
+            userResponse.getPasswordHash(),
```

Comment:
I understand this better now.  Rather than implementing `UserDetails` which is the preferred route, we extended `User`.  User has a domain rule that password cannot be null.  In the refresh context and for the purposes of caching this is irrelevant.  The `UserResponse` from active never gets anywhere near the cache.  Which start me wondering if we need a user cache at all.  Though since we have one I will let that particular issue lie.  It does point out that the timeout in the token and the timeout in the cache might need some sync.  Why eject a user from the cache in 30 minutes if their token can last 12 hours?

I will fix the issue with the refresh token.